### PR TITLE
Implementations of Dependency Injection in C#

### LINKED ISSUE #6659: [Languages] Implementations of dependency injection in C#
#### Proposed title of the article
[Languages] Implementations of dependency injection in C#

#### Proposed article introduction
Dependence is an object that another object relies on for its existence. When an object wants an object, it doesn't have to build it. This is called dependency injection (or inversion of control). Mocking the dependencies is an effective way to make testing more efficient.

The fact that class B calls a method on class C indicates that class A is dependent upon both classes B and C. Instead of requiring these classes to create instances of B and C, we can use dependency injection to provide an instance of class C to class B and an instance of class B to class A.

#### Key takeaways 
The article will cover:
- An overview of dependency injection 
- How to implement dependency injection using method injection in C#
- How to implement dependency injection using constructor Injection C#
- How to implement dependency injection using property injection c#
- How to implement dependency injection using interface-based injections in C#

#### Article quality
 This article would deal with explaining dependency injection and its implementation in C# in detail. To make my article unique I will create my own project to demonstrate the implementation process. Also, my article will be beginner-friendly and the code snippets will be well explained and tested. 
#### References
N/A

### PR DESCRIPTION
**Pull request by djayjames**
#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

#### Topic suggestion this closes

This closes #6659

### CODE REVIEW

**Comment by Kayere** [comment]
Hello @djayjames,

 Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept.

Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.

Be sure to see our resources page to see more info on plagiarism and what is considered such.

[Implementations of dependency injection in C# #7340 Copyleaks Report.pdf](https://github.com/section-engineering-education/engineering-education/files/8368966/Implementations.of.dependency.injection.in.C.7340.Copyleaks.Report.pdf)

**Comment by Dawe-7** [comment]
> Hello @djayjames,
> 
>  Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept.
> 
> Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.
> 
> Be sure to see our resources page to see more info on plagiarism and what is considered such.
> 
> [Implementations of dependency injection in C# #7340 Copyleaks Report.pdf](https://github.com/section-engineering-education/engineering-education/files/8368966/Implementations.of.dependency.injection.in.C.7340.Copyleaks.Report.pdf)

@djayjames, any updates on resolving the plagiarism?

**Comment by Dawe-7** [comment]
@djayjames, kindly check the file path (URL) of your article in VS Code. It could bring about merge conflicts.

**Comment by WanjaMIKE** [comment]
Some of your code seem to appear on other blogs. Please check it in the attached report.
[New Scan 9_07 AM Copyleaks Report.pdf](https://github.com/section-engineering-education/engineering-education/files/8450428/New.Scan.9_07.AM.Copyleaks.Report.pdf)

**Comment by Dawe-7** [comment]
> @djayjames, kindly check the file path (URL) of your article in VS Code. It could bring about merge conflicts.

@djayjames, any updates on fixing this and the plagiarism?

**Comment by Dawe-7** [comment]
> Some of your code seem to appear on other blogs. Please check it in the attached report. [New Scan 9_07 AM Copyleaks Report.pdf](https://github.com/section-engineering-education/engineering-education/files/8450428/New.Scan.9_07.AM.Copyleaks.Report.pdf)

@djayjames, is this article fully polished and ready for review?

**Comment by djayjames** [comment]
@Dawe-7 I have done the changes, please check.

File: content/articles/implementation-of-dependency-injection-in-csharp/index.md [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+Property dependency injection is used when we intend to create the dependency object when it is required.
+
+### Implementing dependency injection using interface-based injections
+Interface-based injection injects a dependency into a client via an interface. The dependency becomes the injector.
+
+An interface-based injection is similar to property injection, using default getter and setter methods. However, interface-based injection sets the interface property using an explicit setter-getter interface.
+
+The following is a c-sharp code implementing property injection:
+
+```c#
```
Comment (Dawe-7) [inline content/articles/implementation-of-dependency-injection-in-csharp/index.md:316]: This code under interface-based injection doesn't seem to run on my end, kindly check for any build errors.

**Review by Dawe-7 — COMMENTED** [review]
Awesome article @djayjames, check and fix the code under which i've left a comment and this article should be ready for final review.

**Comment by djayjames** [comment]
Hello @Dawe-7, I have corrected the code snippet. Please check

**Review by Dawe-7 — APPROVED** [review]
Nice article @djayjames. Kindly go through the changes made throughout the review process so as to improve on the flow and format of your future articles.

**Review by Kayere — APPROVED** [review]
Nice article @djayjames.
Thank you for the review @Dawe-7.

**Pull request merged**